agent.go

```diff
@@ -0,0 +1,89 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ Agents = (*agents)(nil)
+
+// Agents describes all the agent-related methods that the
+// Terraform Cloud API supports.
+// TFE API docs: https://www.terraform.io/docs/cloud/api/agents.html
+type Agents interface {
+	// Read an agent by its ID.
+	Read(ctx context.Context, agentID string) (*Agent, error)
+
+	// List all the agents of the given pool.
+	List(ctx context.Context, agentPoolID string, options *AgentListOptions) (*AgentList, error)
+}
+
+// agents implements Agents.
+type agents struct {
+	client *Client
+}
+
+// AgentList represents a list of agents.
+type AgentList struct {
+	*Pagination
+	Items []*Agent
+}
+
+// Agent represents a Terraform Cloud agent.
+type Agent struct {
+	ID   string `jsonapi:"primary,agents"`
+	Name string `jsonapi:"attr,name"`
+	IP   string `jsonapi:"attr,ip-address"`
+
+	// Relations
+	Organization *Organization `jsonapi:"relation,organization"`
+	Workspaces   []*Workspace  `jsonapi:"relation,workspaces"`
```

Comment:
I think these can be removed because these relations aren't available on agents according to the docs